### PR TITLE
[Sage-512] Sage 3 - Modal Updates

### LINKED ISSUE #512: Update Modal
## Description

Update component to match latest specs in Sage 3.0. The following were noted during initial audit:

- Ensure icon in header is possible
- Add subhead with popover affordance
- Ensure correct border radius 

**Dependencies**:
- [Border radius](https://github.com/Kajabi/sage-lib/issues/472)

**Links**
- [Figma specs](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/WIP-Sage-3-%E2%80%94-Admin-Components?node-id=3818%3A14755)
- [Existing component(s)](https://sage-design-system.kajabi.com/pages/component/modal)

### PR DESCRIPTION
**Pull request by anechol**
## Description

Updates Modal component to Sage 3:
- Adds `subheader` prop to header
- Adds props for Popover title, content, and link
- Adds optional `icon` to header
- Adds missing `image` prop to React component

## Screenshots

Screen Shot 2021-08-30 at 12 56 23 PM

Screen Shot 2021-08-31 at 12 48 38 PM

Screen Shot 2021-08-31 at 12 14 32 PM

### Rails
- View [Rails component](http://localhost:4000/pages/component/modal)
- Check that component UI exists and new attributes are not broken

### React
- View [Storybook](http://localhost:4100/?path=/docs/sage-modal--default)
- Check that component UI exists and new attributes are not broken

## Testing in `kajabi-products`

(MEDIUM) Adds new props for the Modal component and updates Modal header markup to accommodate optional icon and image props.

## Related

Closes #512

### CODE REVIEW

**Review by monicawheeler — APPROVED** [review]
LGTM!

**Review by FuturaExtraBold — APPROVED** [review]
Nice work, Ashley!

File: packages/sage-react/lib/Modal/ModalHeader.jsx [hunk 1]
```diff
@@ -31,12 +66,31 @@ ModalHeader.defaultProps = {
...
   className: PropTypes.string,
+  icon: PropTypes.shape({
+    color: PropTypes.oneOf(Object.values(Icon.COLORS)),
+    name: PropTypes.oneOf(Object.values(Icon.ICONS)),
+  }),
+  image: PropTypes.shape({
+    alt: PropTypes.string,
+    src: PropTypes.string
+  }),
+  popover: PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (anechol) [reply packages/sage-react/lib/Modal/ModalHeader.jsx:88]: Hey, thanks for pointing this out. I didn't consider that as an option. I can absolutely look into this as a follow-up ticket for a refactor.

File: packages/sage-react/lib/Modal/ModalHeader.jsx [hunk 2]
```diff
@@ -18,9 +25,37 @@ export const ModalHeader = ({
...
+              ? SageTokens.GRID_TEMPLATES.ETE
+              : SageTokens.GRID_TEMPLATES.TE
+          }
+        >
+          {icon && <Icon className="sage-modal__header-icon" icon={icon.name} color={icon.color} /> }
+          {image && <img src={image.src} alt={image.alt || ''} />}
+          {title && <h1 className={SageClassnames.TYPE.HEADING_4}>{title}</h1>}
+          {aside && <ModalHeaderAside>{aside}</ModalHeaderAside>}
+        </Card.Row>
+        {subheader && (
```
[the remark this answers is not in the document]
Comment (anechol) [reply packages/sage-react/lib/Modal/ModalHeader.jsx:41]: Same here as below

File: docs/lib/sage_rails/app/views/sage_components/_sage_modal_content.html.erb [hunk 3]
```diff
@@ -1,21 +1,49 @@
...
+            <%= content_for :sage_header_aside %>
+          </aside>
+        <% end %>
+      <% end %>
+      <% if component.subheader %>
+        <%= sage_component SageCardRow, { grid_template: "te" } do %>
+          <h2 class="t-sage-heading-6">
+            <%= component.subheader %>
+          </h2>
+          <% if component.help_title %>
```
[the remark this answers is not in the document]
Comment (anechol) [reply docs/lib/sage_rails/app/views/sage_components/_sage_modal_content.html.erb:29]: For point A, I think that's a question I'd have to ask design about as I'm unsure how that would look. For point B, I think that's a good consideration. Appreciate the insight.

@philschanely Do you mind collecting all your feedback here and creating an issue for me to work through for a future update?

**Pull request merged**